[Bug] Flaky test: TestCreateDataDirectory_ExistingDirectory

There is a flaky test on Windows in `app`:

https://github.com/temporalio/cli/actions/runs/6252251042/job/16975099634?pr=321

```
 error cleaning up temp dir: remove C:\Users\RUNNER~1\AppData\Local\Temp\temporal_test\TestCreateDataDirectory_MissingDirectory\foo\bar\baz.db: The process cannot access the file because it is being used by another process.
2023/09/20 17:56:29 INFO  No logger configured for temporal client. Created default one.
error cleaning up temp dir: remove C:\Users\RUNNER~1\AppData\Local\Temp\temporal_test\TestCreateDataDirectory_ExistingDirectory\foo.db: The process cannot access the file because it is being used by another process.
--- FAIL: TestCreateDataDirectory_ExistingDirectory (2.77s)
```

Seems to fail intermittently on Windows only—I'd guess because something is still holding `foo.db` open during test cleanup. We should figure out what that is and fix it so this test passes consistently.

(I'm guessing this same condition exists on Mac/Linux, too, but we never see errors because you're allowed to delete files while they're still open on UNIXy OSes.)

--- a/app/app_test.go
+++ b/app/app_test.go
@@ -320,7 +320,7 @@ func assertServerHealth(ctx context.Context, t *testing.T, opts sdkclient.Option
 		t.Error(clientErr)
 	}
 
-	// Give the server 10 seconds to become healthy.
+	// Give the server 1 second to become healthy.
 	for i := 0; i < 10; i++ {
 		_, clientErr = c.CheckHealth(ctx, nil)
 		if clientErr == nil {

--- a/app/app_test.go
+++ b/app/app_test.go
@@ -320,8 +320,16 @@ func assertServerHealth(ctx context.Context, t *testing.T, opts sdkclient.Option
 		t.Error(clientErr)
 	}
 
-	if _, err := c.CheckHealth(ctx, nil); err != nil {
-		t.Error(err)
+	// Give the server 10 seconds to become healthy.
+	for i := 0; i < 10; i++ {
+		_, clientErr = c.CheckHealth(ctx, nil)
+		if clientErr == nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	if clientErr != nil {
+		t.Error(clientErr)
 	}
 
 	// Check for pollers on a system task queue to ensure that the worker service is running.

--- a/server/commands.go
+++ b/server/commands.go
@@ -337,6 +337,7 @@ func NewServerCommands(defaultCfg *sconfig.Config) []*cli.Command {
 				if err := s.Start(); err != nil {
 					return cli.Exit(fmt.Sprintf("Unable to start server. Error: %v", err), 1)
 				}
+				s.Stop()
 				return cli.Exit("All services are stopped.", 0)
 			},
 		},

